cmake/Format.cmake

```diff
@@ -0,0 +1,67 @@
+# Returns a list of all files that has been changed in current branch compared
+# to master branch. This includes unstaged, staged and committed files.
+function(get_changed_files outvar)
+  set(default_branch master)
+
+  execute_process(
+    COMMAND git branch --show-current
+    OUTPUT_VARIABLE current_branch
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  execute_process(
+    COMMAND git merge-base ${default_branch} ${current_branch}
+    OUTPUT_VARIABLE ancestor_commit
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  # Changed files that have been committed
+  execute_process(
+    COMMAND git diff --name-only ${ancestor_commit}...${current_branch}
+    OUTPUT_VARIABLE committed_files
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  separate_arguments(committed_files NATIVE_COMMAND ${committed_files})
+
+  # Unstaged files
+  execute_process(
+    COMMAND git diff --name-only
+    OUTPUT_VARIABLE unstaged_files
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  separate_arguments(unstaged_files NATIVE_COMMAND ${unstaged_files})
+
+  # Staged files
+  execute_process(
+    COMMAND git diff --cached --name-only
+    OUTPUT_VARIABLE staged_files
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  separate_arguments(staged_files NATIVE_COMMAND ${staged_files})
+
+  set(files ${committed_files} ${unstaged_files} ${staged_files})
+  list(REMOVE_DUPLICATES files)
+
+  set(${outvar} "${files}" PARENT_SCOPE)
+endfunction()
+
+get_changed_files(changed_files)
+
+if(LANG STREQUAL c)
```

Comment:
Similar to the discussion for the lint targets, I'd suggest that Format.cmake just takes a list of files instead of putting a switch-case in here for all possible targets. But passing files is unpleasant with `-P`, so I guess that's the tradeoff. Do we need `-P` though?

Not a blocker anyways.

Reply:
I know, I tried to think of a non-`-P` solution for this. Since calling functions from targets is impossible cmake practically forces you to use `-P`, or to suffer a configure time penalty for all targets. It's possible to avoid `-P` if we simply format all files, but that takes unacceptable amount of time due to uncrustify.

Sorry, I can't think of a cleaner solution for this at the moment. I'll revisit this later and see if I can come up with something simpler.